File: server/src/main/java/sageassistant/model/TobeReceive.java
/*
 * @Author         : Robert Huang<56649783@qq.com>
 * @Date           : 2022-03-26 17:01:30
 * @LastEditors    : Robert Huang<56649783@qq.com>
 * @LastEditTime   : 2022-09-22 14:04:29
 * @FilePath       : \server\src\main\java\sageassistant\model\TobeReceive.java
 * @CopyRight      : Dedienne Aerospace China ZhuHai
 */
package sageassistant.model;

import java.math.BigDecimal;
import java.util.Date;
import lombok.Getter;
import lombok.Setter;
import sageassistant.model.base.ModelTemplate;

@Getter
@Setter
public class TobeReceive extends ModelTemplate {

    private static final long serialVersionUID = 1L;

    private String ProjectNO;
    private String PurchaseNO;
    private String Line;
    private String VendorCode;
    private String VendorName;
    private String PN;
    private Integer Qty;
    private String Unit;
    private String Description;
    private BigDecimal NetPrice;
    private String Currency;
    private BigDecimal USD;
    private Float Rate;
    private Date OrderDate;
    private Date AckDate;
    private Date ExpectDate;
    private String CreateUser;
}

File: web/src/mock/TobeReceive.js
import Mock from 'mockjs'

Mock.setup({
  timeout: '1000-3000'
})

Mock.mock(RegExp('^(/Data/TobeReceive)' + '.*'), options => {
  // list = {data:[{},{}]}
  let list = Mock.mock({
    'data|1-100': [
      {
        'ProjectNO': /(Z|H|S|T|M)CC\d{6}-\d{1,2}/,
        'PurchaseNO': /(Z|H|S|T|M)CF\d{6}/,
        'Line|1-100': 1,
        'VendorCode': /0000\d{ 1}/,
        'VendorName': () => { return Mock.mock('@sentence(3, 5)') },
        'PN': /([A-Z]){6}/,
        'Qty|1-100': 1,
        'Unit': 'UN',
        'Description': () => { return Mock.mock('@sentence(3, 5)') },
        'NetPrice|1000-2000': 1,
        'Currency': /(RMB|USD|EUR|HKD)/,
        'RMB|10-200000': 1,
        'Rate|6-9': 1,
        'AckDate': () => { return Mock.mock('@date("yyyy-MM-dd")') },
        'ExpectDate': () => { return Mock.mock('@date("yyyy-MM-dd")') },
        'OrderDate': () => { return Mock.mock('@date("yyyy-MM-dd")') },
        'CreateUser': /0000[A-D]{1}/
      }
    ]
  })
  return list.data
})
